[docs] Push notifications changes from FCMv1 work (#29921)
expo/expo: docs/pages/versions/unversioned/sdk/notifications.mdx

| false                              |          Killed          | none                                                                    |
| true                               |        Foreground        | `NotificationReceivedListener` & `NotificationResponseReceivedListener` |
| true                               |        Background        | `NotificationResponseReceivedListener`                                  |
| true                               |          Killed          | `useLastNotificationResponse`                                           |

In the table above, whenever `NotificationResponseReceivedListener` is triggered, the same would apply to the `useLastNotificationResponse` hook.

> > **info** When the app is not running or killed, and is restarted by tapping on a notification, your `NotificationResponseReceivedListener` may or may not be triggered. To reliably capture the response, we recommend using `useLastNotificationResponse`. 

Reviewer: ```suggestion
> **info** When the app is not running or killed and is restarted by tapping on a notification, the `NotificationResponseReceivedListener` may or may not be triggered. To reliably capture the response, we recommend using `useLastNotificationResponse`. 
```

### Background notifications

> **warning** Background event listeners are not supported in Expo Go.

A background notification (iOS) or a data-only notification (Android) is a remote notification that does not display an alert, play a sound, or add a badge to your app's icon. The purpose of a background notification is to provide a way to wake up your app to trigger an app data refresh in the background.

To handle notifications while the app is in the background or not running, you need to do the following:

- Add `expo-task-manager` package to your project.

> - In your application code, set up a [background task](/versions/unversioned/sdk/notifications/#registertaskasynctaskname) to run when the notification is received.

Reviewer: ```suggestion
- In your application code, set up a [background task](/versions/latest/sdk/notifications/#registertaskasynctaskname) to run when the notification is received.
```

> For iOS:

Reviewer: ```suggestion
For Android:

- Send a push notification payload containing only the `data` key.

For iOS:
```

Let's add the Android part before iOS as per the [Platform order guideline in the writing style guide(https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#referencing-android-ios-and-web).

> - In the array under `ios.infoPlist.UIBackgroundModes` key in your [app config](/versions/latest/config/app/), add the values `remote-notification` and `processing`.

Reviewer: ```suggestion
- In the array under the `ios.infoPlist.UIBackgroundModes` key in your [app config](/workflow/configuration/), add the values `remote-notification` and `processing`.
```

The link here should be that of app config guide than the API reference since we're not linking the property directly from the app config reference.

- Add `_contentAvailable: true` to your push notification payload for the Expo push notification service. Under normal circumstances, the "content-available" flag should launch your app in the background.

> 
> For Android:

Reviewer: ```suggestion
```

See above comment about platform order.

> - Send a push notification payload containing only the `data` key.

Reviewer: ```suggestion
```
